### PR TITLE
Basic app skeleton + layout examples

### PR DESCRIPTION
**Pull request by zhengmar3**
- Basic app setup with Vite.js, MUI, react-query, react-router-dom, tailwindcss and mantine-hooks.
- Added very basic responsive header+navbar+main layout.
- Added simple layout examples (holy grail, sidebar, simple layout)

Very basic, but want to get some feedback on the overall project structure hence the PR.

On >md mediaBreakPoints:

Screenshot 2022-08-25 at 16 03 22

On <sm mediaBreakPoints:

Screenshot 2022-08-25 at 16 03 58
Screenshot 2022-08-25 at 16 03 42

### CODE REVIEW

**Review by yarivus — COMMENTED** [review]
@conor-r3 please review in detail ;)

File: mui-template-app/src/components/AppRoutes/AppRoutes.tsx [hunk 1]
```diff
@@ -3,13 +3,20 @@ import { Routes, Route } from "react-router-dom";
...
+import SidbarLayout from "../SidbarLayout/SidbarLayout";
 
 function AppRoutes() {
   return (
     <Routes>
       <Route path="/" element={<AppShell />}>
         <Route index element={<Dashboard />} />
-        <Route path="layout" element={<Layout />} />
+        <Route path="layout" element={<Layout />}>
+          <Route path="holygrail" element={<HolyGrailLayout />} />
```
Comment (yarivus) [inline mui-template-app/src/components/AppRoutes/AppRoutes.tsx:16]: fix indent

File: mui-template-app/src/components/SidbarLayout/SidbarLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+function SidbarLayout() {
```
Comment (yarivus) [inline mui-template-app/src/components/SidbarLayout/SidbarLayout.tsx:1]: Sidbar => Sidebar

**Review by yarivus — CHANGES_REQUESTED** [review]

File: mui-template-app/src/components/SidbarLayout/SidbarLayout.tsx (hunk 2 above)
Comment (zhengmar3) [reply mui-template-app/src/components/SidbarLayout/SidbarLayout.tsx]: Oh fixed

File: mui-template-app/src/components/AppRoutes/AppRoutes.tsx (hunk 1 above)
Comment (zhengmar3) [reply mui-template-app/src/components/AppRoutes/AppRoutes.tsx:16]: I think this is in correct indentation.

_<Route path="holygrail" element={<HolyGrailLayout />} />_  is actually a child node of _<Route path="layout" element={<Layout />}>_

Hence the indentation.

Also, I have visual studio prettier plugin setup globally automatically. So the indentation should done automatically.

**Review by yarivus — COMMENTED** [review]
LGTM

File: mui-template-app/src/components/AppRoutes/AppRoutes.tsx (hunk 1 above)
Comment (ellitrym1) [reply mui-template-app/src/components/AppRoutes/AppRoutes.tsx:16]: am i correct in assuming that the actual path for holygrail would be "/layout/holygrail" hence the indentation?

**Review by yarivus — COMMENTED** [review]
@conor-r3 please review this.
@zhengmar3 when can we see a demo ?

**Comment by zhengmar3** [comment]
@yarivus 

This is just a first PR for this professional service starter-template project.
No deployment is setup for this yet. But since it's pure front-end project. It's super easy to run the code locally.

If you want to see how it looks in browser, just git clone the code and:

``cd mui-template-app``
``npm install``
``npm run dev``

Otherwise, I will try to demo this work-in-progress project through all the front-end show and tell meeting.

**Comment by yarivus** [comment]
@zhengmar3 thanks 😉
I've tried it out but obviously there's not much yet ...
I'll try in the next PRs when it has more stuff.
I do encourage to hold a quick demo and seek feedback in the next FE show & tell.

File: mui-template-app/src/components/Layout/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import Stack from "@mui/material/Stack";
+import PageHeader from "../PageHeader/PageHeader";
+import { Link, Outlet } from "react-router-dom";
+import Button from "@mui/material/Button";
+import Box from "@mui/material/Box";
+import Divider from "@mui/material/Divider";
+
+function Layout() {
```
Comment (conor-r3) [inline mui-template-app/src/components/Layout/Layout.tsx:8]: This is a good idea!

**Pull request merged**